ibis/common/geospatial.py

```diff
@@ -13,28 +13,54 @@
 except ImportError:
     ...
 
+NumberType = TypeVar('NumberType', int, float)
+PointType = Iterable[NumberType]
+LinestringType = Iterable[PointType]
+PolygonType = Iterable[LinestringType]
+MultipolygonType = Iterable[PointType]
 
-def _format_point_value(value: Iterable) -> str:
+
+def _format_point_value(value: PointType) -> str:
     """Convert a iterable with a point to text."""
     return ' '.join(str(v) for v in value)
 
 
-def _format_linestring_value(value: Iterable) -> str:
+def _format_linestring_value(value: LinestringType, nested=False) -> str:
     """Convert a iterable with a linestring to text."""
-    return ', '.join(_format_point_value(point) for point in value)
+    template = '({})' if nested else '{}'
+    if not isinstance(value[0], (tuple, list)):
+        msg = ' data structure expected: Iterable[Iterable[Number]]'
+        msg = ('Linestring' if not nested else 'Inner') + msg
+        raise ex.IbisInputError(msg)
+    return template.format(
+        ', '.join(_format_point_value(point) for point in value)
+    )
 
 
-def _format_polygon_value(value: Iterable) -> str:
+def _format_polygon_value(value: PolygonType, nested=False) -> str:
     """Convert a iterable with a polygon to text."""
-    return ', '.join(
-        '({})'.format(_format_linestring_value(line)) for line in value
+    template = '({})' if nested else '{}'
+    if not isinstance(value[0][0], (tuple, list)):
+        msg = ' data structure expected: Iterable[Iterable[Iterable[Number]]]'
+        msg = ('Polygon' if not nested else 'Inner') + msg
```

Comment:
I think it's a bit easier to read if you use a format string here.

Reply:
Ok I will change that! thanks!